src/Peer.cpp

```diff
@@ -57,17 +57,44 @@ int Peer::getFd(void) const
 	return this->_fd;
 }
 
+void	Peer::setUsername(const std::string &username)
+{
+	if (this->_username.empty())
+		this->_username = username;
+}
+
+void	Peer::setMode(const std::string &mode)
+{
+	if (this->_mode.empty())
+		this->_mode = mode;
```

Comment:
are we sure we want to be unable to change mode once set ?

Reply:
It is to avoid re-setting the fields. Mode will be discarded soon anyway...